standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java

```diff
@@ -1873,8 +1873,55 @@ public enum ConfVars {
             " positive value will be used as-is."
     ),
     CATALOG_SERVLET_AUTH("metastore.catalog.servlet.auth",
-        "hive.metastore.catalog.servlet.auth", "jwt", new StringSetValidator("none", "simple", "jwt"),
-        "HMS Catalog servlet authentication method (none, simple, or jwt)."
+        "hive.metastore.catalog.servlet.auth", "jwt", new StringSetValidator("none", "simple", "jwt", "oauth2"),
+        "HMS Catalog servlet authentication method (none, simple, jwt, or oauth2)."
+    ),
+    CATALOG_SERVLET_AUTH_OAUTH2_ISSUER("metastore.catalog.servlet.auth.oauth2.issuer",
+        "hive.metastore.catalog.servlet.auth.oauth2.issuer", "",
+        "The issuer(iss)'s URI. This is required when you use metastore.catalog.servlet.auth=oauth2"
+    ),
+    CATALOG_SERVLET_AUTH_OAUTH2_VALIDATION_METHOD("metastore.catalog.servlet.auth.oauth2.validation.method",
```

Comment:
should we define token type instead:
````
There are two types of access tokens, depending on how they are encoded:

Identifier-based: The token represents a random, hard-to-guess identifier associated with the authorization in the authorization server’s database.

Self-contained: The authorization is encoded within the token itself and is protected through cryptography to prevent tampering. JSON Web Token (JWT) is the common standard for this method.

For self-contained tokens, the authorization-related metadata can be directly parsed from the access token. However, for identifier-based tokens, the authorization server’s token introspection functionality must be used to validate/retrieve the metadata.
````

Reply:
Precisely, the format of a token is not tightly coupled with the validation method. I know two use cases.
(1) Some authorization servers can allow end users to revoke existing integrations. JWT can't express such a revocation. Resource servers can use the introspection endpoint to know the latest status.
(2) [Some authorization servers may not want to include all information in JWT](https://www.keycloak.org/docs/latest/server_admin/#_using_lightweight_access_token), not to expose PII or to reduce the token size. A token introspection response can contain more details than a JWT.